tests/absolute_helpers/mod.rs

```diff
@@ -0,0 +1,197 @@
+where
+    P: AsRef<path::Path>,
+    Q: AsRef<path::Path>,
+{
+    #[cfg(windows)]
+    {
+        use std::os::windows::fs as winfs;
+
+        let dst = dst.as_ref();
+
+        winfs::symlink_dir(src, &dst).expect(
+            "Could not create symbolic link. \
+             Run as Administrator, or on Windows 10 in Developer Mode",
+        );
+        dst.symlink_metadata().expect(
+            "Link creation succeeded, but can't read link? \
+             If you're using Wine, see bug 44948",
+        );
+    }
+
+    #[cfg(unix)]
+    {
+        use std::os::unix::fs as unixfs;
+
+        unixfs::symlink(src, dst).expect("Could not create symbolic link");
+    }
+
+    #[cfg(all(not(windows), not(unix)))]
+    unreachable!();
+}
+
+#[test]
+fn absolute_path_is_idempotent() {
+    ::setup();
+    // The current_dir() result is always absolute,
+    // so absolutizing it should not change it.
+
+    let actual = PathArc::new(env::current_dir().unwrap())
+        .absolute()
+        .unwrap();
+    let expected = env::current_dir().unwrap().canonicalize().unwrap();
+
+    assert_eq!(actual.as_path(), expected.as_path());
+}
+
+#[test]
+fn absolute_path_removes_currentdir_component() {
+    ::setup();
+    let actual = PathArc::new("foo/./bar").absolute().unwrap();
+    let expected = PathArc::new("foo/bar").absolute().unwrap();
+
+    assert_eq!(actual, expected);
+}
+
+#[test]
+fn absolute_path_removes_empty_component() {
+    ::setup();
+    let actual = PathArc::new("foo//bar").absolute().unwrap();
+    let expected = PathArc::new("foo/bar").absolute().unwrap();
+
+    assert_eq!(actual, expected);
+}
+
+#[test]
+fn absolute_path_lexically_resolves_parentdir_component() {
+    ::setup();
+    let tmp_dir = TempDir::new("normalize_parentdir").unwrap();
+    let a_dir = tmp_dir.path().join("a");
+    fs::create_dir_all(&a_dir).unwrap();
+
+    let b_dir = tmp_dir.path().join("b");
+    fs::create_dir_all(&b_dir).unwrap();
+
+    fs::create_dir_all(&b_dir.join("target")).unwrap();
+
+    let link_path = a_dir.join("link");
+    symlink_dir("../b/target", link_path);
+
+    // Because of the symlink, a/link/../foo is actually b/foo, but
+    // lexically resolving the path produces a/foo.
+    let actual = PathArc::new(a_dir.join("link/../foo")).absolute().unwrap();
+    let expected = PathArc::new(a_dir.join("foo")).absolute().unwrap();
+
+    assert_eq!(actual, expected);
+}
+
+#[test]
+fn absolute_path_interprets_relative_to_current_directory() {
+    ::setup();
+    let actual = PathArc::new("foo").absolute().unwrap();
+    let expected = PathArc::new(env::current_dir().unwrap())
+        .join("foo")
+        .absolute()
+        .unwrap();
+
+    assert_eq!(actual, expected);
+}
+
+#[cfg(unix)]
+mod unix {
+    use super::*;
+
+    #[test]
+    fn absolute_path_need_not_exist() {
+        ::setup();
+        let path = PathArc::new("/foo/bar").absolute().unwrap();
+
+        assert_eq!(path.as_os_str(), "/foo/bar");
```

Comment:
as unlikely as it is for `/foo/bar` to exist, I would like this to be renamed something like: `/this_path/does_not_exist` as that is much MORE likely to not exist :)

Reply:
Changed to `/does/not/exist` and added code to verify that it doesn't actually exist.